metric/metrics.go

```diff
@@ -24,25 +25,41 @@ const (
 )
 
 // Set is the basic structure for storing metrics.
-type Set map[string]interface{}
+type Set struct {
+	// The storer is required for deltas and rates, but forces us overriding marshaller and makes unmarshaller
```

Comment:
I don't get the point about `makes unmarshaller impossible.`. The same way we implement the `MarshalJSON` it could implement `UnmarshalJSON`. I'm missing something?

Reply:
We could rebuild the `Metrics` map from a stored JSON, but even if it's technically possible, I don't think it would be a good Idea to build the `Storer` instance from a stored JSON (since this configuration could change).